### PR TITLE
Update Nessie to 0.71.0

### PR DESCRIPTION
**Pull request by wendigo**
I didn't update to newest version as it relies on the newest jackson which will be updated soon in Trino.

### CODE REVIEW

**Comment by nastra** [comment]
Note that this is being updated as part of https://github.com/trinodb/trino/pull/19434.

**Comment by wendigo** [comment]
@nastra can we split this change? it seems that Nessie update is independent of the Iceberg update

**Comment by nastra** [comment]
@wendigo I don't think this should be separated, because the Nessie version should be matching the one used by Iceberg.

**Comment by wendigo** [comment]
@nastra How we can check/enforce it then? It seems that if I upgrade nessie version  by mistake without upgrading Iceberg tests are passing

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java [hunk 1]
```diff
@@ -46,9 +46,9 @@ protected void setup(Binder binder)
     public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalogConfig icebergNessieCatalogConfig)
     {
         return new NessieIcebergClient(
-                HttpClientBuilder.builder()
+                NessieClientBuilder.createClientBuilder(null, null)
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java:49]: javadoc of `HttpClientBuilder.builder()` suggests migrating to `NessieClientBuilder#createClientBuilderFromSystemSettings()`. Why do we use `createClientBuilder()` method?

**Pull request closed without merging**